### PR TITLE
Dependencies update for C# driver.

### PR DESCRIPTION
**Pull request by farost**
## What is the goal of this PR?

We add all the needed dependencies for building the new C# driver for TypeDB.

## What are the changes implemented in this PR?

This PR contains bazel's csharp rules' pulling commands and additional bazel rules for building the native C# library used in the driver and its automatic tests. 

This PR **will be updated** after we finish the deployment preparations work on later stages of the development, so there are still some TODOs left. The current state of this PR is just needed for referencing from the main [C# driver's PR](https://github.com/vaticle/typedb-driver/pull/609). 

I've put a **status: blocked** label for explicitness, but I'm not sure what's the correct process, so feel free to correct me.

### CODE REVIEW

**Comment by farost** [comment]
Self review incoming in several hours.

File: builder/swig/csharp.bzl
Comment (farost) [inline builder/swig/csharp.bzl:18]: Not sure if it's still relevant. There are some similarities, but such refactoring could make the readability worse. Will think on the deployment stage.

File: builder/swig/csharp.bzl
Comment (farost) [inline builder/swig/csharp.bzl:82]: This is a useful technique to save separate sets of files for future usage. I had troubles with separating C# and CXX files from each other to compile both source and test libraries, and it was a perfect solution!

File: builder/swig/csharp.bzl
Comment (farost) [inline builder/swig/csharp.bzl:139]: We need to put DotnetAssembly*Info to build our test (specifically BDD test) targets, which require this information for dependencies.

File: builder/swig/csharp.bzl
Comment (farost) [inline builder/swig/csharp.bzl:201]: Will make an additional PR for it.

File: builder/swig/csharp.bzl
Comment (farost) [inline builder/swig/csharp.bzl:192]: Will check it out in an additional PR for deployment.

File: builder/csharp/deps.bzl
Comment (dmitrii-ubskii) [inline builder/csharp/deps.bzl:26]: ⛔️

File: WORKSPACE
Comment (dmitrii-ubskii) [inline WORKSPACE:57]: Do we actually have any .NET targets in dependencies? This should only appear in the repos that do.

File: builder/swig/csharp.bzl
Comment (dmitrii-ubskii) [inline builder/swig/csharp.bzl:156]: * How many of these empty args are defaulted?
* `"{}".format(...)` does nothing, `File.basename` is already a `string`

**Review by dmitrii-ubskii — APPROVED** [review]
Generally approved provided the couple minor comments are addressed

**Comment by dmitrii-ubskii** [comment]
Looks like you have some unsigned commits. You'll want to amend them with a signature before you merge.

File: builder/csharp/deps.bzl
Comment (farost) [reply builder/csharp/deps.bzl:26]: Fixed!

File: WORKSPACE
Comment (farost) [reply WORKSPACE:57]: Looks like you're right, we don't need it here.

File: builder/swig/csharp.bzl
Comment (farost) [reply builder/swig/csharp.bzl:156]: * Not a single one, I was inserting it while getting errors.
* I know, but it looked better for me to have all the rules names in a similar format with .format. I can remove it if it disturbs a reader too much.

Comment (farost) [reply builder/swig/csharp.bzl:156]: Approved irl

**Pull request merged**
